vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -689,21 +689,39 @@ def build_connector_meta(
                 kv_transfer_params=req.kv_transfer_params,
             )
 
-        for req_id, (req, block_ids) in self._reqs_need_save.items():
+        # NOTE: For prefill side, there might be chance that early added
+        # request is chunked prefill, so we need to check if new blocks are added
+        for req_id, new_block_id_groups, preempted in yield_req_data(scheduler_output):
+            if req_id not in self._reqs_need_save:
+                continue
+            if preempted:
+                self._reqs_need_save.pop(req_id)
+            if new_block_id_groups is None:
+                continue
+
+            req, _ = self._reqs_need_save[req_id]
             assert req.kv_transfer_params is not None
             meta.add_new_req_to_save(
                 request_id=req_id,
-                local_block_ids=block_ids,
+                local_block_ids=new_block_id_groups[0],
                 kv_transfer_params=req.kv_transfer_params,
             )
+            # NOTE: for partial prefill, we will keep the req_id until the full
+            # prefill is done. Clear once transfer on full prefill is started.
+            assert scheduler_output.num_scheduled_tokens is not None
+            num_scheduled_tokens = scheduler_output.num_scheduled_tokens[req_id]
+            is_partial = (
+                req.num_computed_tokens + num_scheduled_tokens
+            ) < req.num_prompt_tokens
```

Comment:
This might work...
But I think it's safer to just move `self._reqs_need_save.pop(req_id)` to `request_finished`.
@NickLucche WDYT?